controllers/gitrepository_controller.go

```diff
@@ -206,10 +207,10 @@ func (r *GitRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Reconcile actual object
 	reconcilers := []gitRepositoryReconcileFunc{
-		r.reconcileStorage,
 		r.reconcileSource,
 		r.reconcileInclude,
 		r.reconcileArtifact,
+		r.reconcileStorage,
```

Comment:
@aryan9600 moved this as the last operation, as reconcileSource is no longer bound to download a new artifact every time. Do you see any challenges with this approach?

Reply:
Any particular reason for doing this that'd benefit the overall change?
This will affect status condition set at the beginning of the reconciliation when we check if the storage is missing artifact and add reconciling condition.

Reply:
I think initially it was because garbage collection was removing the valid artefact from the storage. But now I believe that was only happening as the `reconcileArtifact` was resulting in new files without names. Will revert this now based on your feedback on that front.